client/src/entry/admin/AdminRouter.js

```diff
@@ -146,16 +144,13 @@ export const getAdminRouter = (Galaxy, options) => {
             this._display_vue_helper(ResetMetadata);
         },
 
-        show_data_manager: function (path) {
-            const Galaxy = getGalaxyInstance();
-            console.log("show_data_manager");
-            const vueMount = document.createElement("div");
-            this.page.display(vueMount);
-            // always set the route back to the base, i.e.
-            // `${galaxyRoot}admin/data_manager`
-            Galaxy.debug("show_data_manager: path='" + path + "'");
-            DataManagerRouter.replace(path || "/");
-            new Vue({ router: DataManagerRouter, render: (h) => h(DataManagerView) }).$mount(vueMount);
+        // Because this has a router in it, we need to be careful about destroying it properly
+        dataManagerInstance: null,
```

Comment:
What does this do ? Is that related to some weird JS scoping I don't know about ?

Reply:
If you take out the destroy and put a console.log in one of the created() hooks in one of the router views, then repeatedly click on the link that mounts the router, it will make multiple instances of the vue instances from the router definitions, many of which have ajax loads.

In order for those "inactive" views to be properly destroyed you must initiate the destroy event on the container instead of just removing the element.

Reply:
So this problem is unique to the fact that the component being mounted "DataManager" contains a vue-router, and the element on which that whole Vue subtree is being mounted is simply being removed from the DOM by backbone when you click this link. (Hope that makes it clear.)